.github/workflows/e2e_tests.yaml

```diff
@@ -45,21 +48,13 @@ jobs:
           key: ${{ runner.os }}-gradle-v2-${{ hashFiles('**/*.gradle*') }}
           restore-keys: ${{ runner.os }}-gradle-v2
       - uses: actions/cache@69d9d449aced6a2ede0bc19182fadc3a0a42d2b0
+        name: AVD Cache
         id: avd-cache
         with:
           path: |
             ~/.android/avd/*
             ~/.android/adb*
           key: avd
-      - name: Generate AVD snapshot for caching
```

Comment:
I am surprised this yields a positive result! Really cool if so.

What I observed on multiple runs where the AVD cache was pre-generated (before API33 at least...) was lower performance on the first run (obviously) but if the AVD cache was pushed on main branch only so it wasn't evicted from github's cache, subsequent runs on main and all PR branches were much faster as a performance benefit

What I further observed was that the emulator performs lots of "first run" type things on it's first boot, so when I gave the emulator time to do that during AVD cache generation - meaning it was already "warm" for cache hits, I actually had greater stability as well, since there were some "first run" type things that failed sometimes (Zygote initial memory setting, for instance, or just race conditions with test operations) that resulted in better AVD stability on cache-hit runs / less flakes, as a stability benefit

If your new CI AVD strategy results in similar or better performance and stability, I need to look at this in react-native-firebase